litgpt/data/json.py

```diff
@@ -103,3 +102,34 @@ def val_dataloader(self) -> DataLoader:
             num_workers=self.num_workers,
             collate_fn=get_sft_collate_fn(max_seq_length=self.max_seq_length, ignore_index=self.ignore_index)
         )
+
+    def get_splits(self) -> Tuple:
+        # A single file (gets split into train and test)
+        if self.json_path.is_file():
+            data = load_split(self.json_path)
+
+            # Partition the dataset into train and test
+            train_data, test_data = random_split(
+                data,
+                [1.0 - self.test_split_fraction, self.test_split_fraction],
+                generator=torch.Generator().manual_seed(self.seed)
+            )
+            return train_data, test_data
+
+        # A directory containing train.json and val.json (or test.json)
+        if (self.json_path / "train.json").is_file():
+            for eval_split in ("val", "test"):
```

Comment:
Yeah, the fact that we have it named test_dataset and `test_split_fraction` is not because I invented that but because it was like that in lit-gpt from the beginning 😄. I can drop `test.json` for now, but further renamings will be required.